examples/basic.zig

```diff
@@ -21,23 +21,28 @@ fn afterAllHook() void {
     std.debug.print("Finished benchmarking.\n", .{});
 }
 
-fn myBenchmark(_: *zbench.Benchmark) void {
+fn myBenchmark(_: std.mem.Allocator) void {
     _ = helloWorld();
 }
 
 test "bench test basic" {
-    const resultsAlloc = std.ArrayList(zbench.BenchmarkResult).init(test_allocator);
-    var benchmarkResults = zbench.BenchmarkResults.init(resultsAlloc);
-    defer benchmarkResults.deinit();
-    var bench = try zbench.Benchmark.init("My Benchmark", test_allocator, .{
+    const stdout = std.io.getStdOut().writer();
+    var bench = zbench.Benchmark.init(test_allocator, .{});
+    defer bench.deinit();
+
+    try bench.add("My Benchmark", myBenchmark, .{
         .iterations = 10,
-        .display_system_info = true,
         .hooks = .{
-            .beforeAll = beforeAllHook,
-            .afterAll = afterAllHook,
+            .before_all = beforeAllHook,
+            .after_all = afterAllHook,
         },
     });
 
-    try zbench.run(myBenchmark, &bench, &benchmarkResults);
-    try benchmarkResults.prettyPrint();
+    const sysinfo = try bench.getSystemInfo();
+    defer sysinfo.deinit();
+    try std.fmt.format(stdout, "\n{}\n", .{sysinfo});
+
+    const results = try bench.run();
+    defer results.deinit();
+    try results.prettyPrint(stdout, true);
```

Comment:
Do we still want to give the end-user this extra line of code? For this PR, i think this is out of the scope. But my preference is that we tackle that with the  
custom output config.

https://github.com/hendriknielaender/zBench/issues/40
```zig
var bench = zbench.Benchmark.init(test_allocator, .{});
defer bench.deinit();

try bench.add("My Benchmark", myBenchmark, .{
  .iterations = 10,
  .hooks = .{
    .before_all = beforeAllHook,
    .after_all = afterAllHook,
  },
});
const results = try bench.run();
defer results.deinit();
```

Reply:
Which extra line of code do you mean? The `results.deinit()`? I suppose it could be freed by `bench.deinit()` if we like. I tended to the results' memory being independently freed because when getting incremental results you likely want to free them after printing them out so that memory usage stays relatively constant. An alternative approach to that might be the results automatically get freed when you call the iterator again so you need to explicitly do a deep copy to keep the results around, that might be the better approach, this way just seemed more standard.

The prettyPrint line was in the old `examples/basic.zig` so I guessed you didn't mean that.
```zig
test "bench test basic" {
    const resultsAlloc = std.ArrayList(zbench.BenchmarkResult).init(test_allocator);
    var benchmarkResults = zbench.BenchmarkResults.init(resultsAlloc);
    defer benchmarkResults.deinit();
    var bench = try zbench.Benchmark.init("My Benchmark", test_allocator, .{
        .iterations = 10,
        .display_system_info = true,
        .hooks = .{
            .beforeAll = beforeAllHook,
            .afterAll = afterAllHook,
        },
    });

    try zbench.run(myBenchmark, &bench, &benchmarkResults);
    try benchmarkResults.prettyPrint();
}
```

Reply:
Reading more closely, I think you might mean you'd like to avoid needing the `prettyPrint` line? Personally I think it's a mistake for a library to choose the writer, the client might have any reason to choose a writer other than stdout. And as far as I know there's no way to supply a writer in the `Benchmark.init()` function and store it to use it in the `run()` function since it's passed as `anytype`, but that might be my ignorance, if there is that'd be nice to pass as an optional config setting. Otherwise the simplest way to let a user choose the writer they want is just to pass it as a parameter to this `prettyPrint()` function.

Also, they might want to write the standard pretty format to stdout and write a different format to a file. IMHO, the most important thing is to give the client access to all the data they need to do whatever they need to, rather than trying to package something for every use case. If they get a `Result` value with all the timing measurements in an array then they can do anything, any kind of analysis or output format they want, that was the main need I was thinking of when writing this version of `Result`.

Reply:
Sorry, not to mentioned it in the first place directly that i meant `prettyPrint()`. Agree and understood your points you made. Since we have an issue discussion open about the plugin system topic. And this seems to be out of scope of this PR, lets move on.